### DISCUSSION TITLE
Models that run in Radioss will not run in OpenRadioss

### DISCUSSION THREAD

**Discussion by marcoserra68** [OpenRadioss/OpenRadioss#2946] [Q&A] [answered]
Hello everyone,

This is just a general question on the differences between Radioss and OpenRadioss. I was not able to find any discussion about this, so here goes:

I have a model that runs perfectly well on Radioss (4 cores). If I take that same model, and run it using OpenRadioss, it crashes on the first step (does go through starter successfully and both are 2024 versions). This is true regardless the number of cores I use in OpenRadioss. Since I have even tried using the same files in the same directory - just pointing to the alternate solver - I can only come to the conclusion that there must be some difference between Radioss and OpenRadioss and the way they work with the model (the .out files seem identical up to the crash though...). One thing that has caught my eye is the fact that there seem to be components used by Radioss that are not found in the OpenRadioss directory. These are things located through the setting of environment variables:

For Radioss...

export RADIOSS_PATH=/opt/altair
export RADFLEX_PATH=$RADIOSS_PATH/hwsolvers/common/bin/linux64
export RAD_CFG_PATH=$RADIOSS_PATH/hwsolvers/radioss/cfg
export RAD_H3D_PATH=$RADIOSS_PATH/hwsolvers/radioss/lib/linux64
export HWSOLVERS_TEMPLEX=$RADIOSS_PATH/hwsolvers/common/bin/linux64
export ABF_PATH=$RADIOSS_PATH/hwsolvers/radioss/bin/linux64
export KMP_STACKSIZE=1000m
export LD_LIBRARY_PATH=$RADIOSS_PATH/hwsolvers/common/bin/linux64/:$RADIOSS_PATH/hwsolvers/radioss/lib/linux64:$LD_LIBRARY_PATH

For OpenRadioss...

export OPENRADIOSS_PATH=/opt/OpenRadioss2024
export RAD_CFG_PATH=$OPENRADIOSS_PATH/hm_cfg_files
export RAD_H3D_PATH=$OPENRADIOSS_PATH/extlib/h3d/lib/linux64
export OMP_STACKSIZE=1000m
export LD_LIBRARY_PATH=$OPENRADIOSS_PATH/extlib/hm_reader/linux64/:$LD_LIBRARY_PATH

Does OpenRadioss not use Templex or Radflex? Does it need them? Why this difference if the two are the same? I have run large models successfully on OpenRadioss, generated both with Hypermesh as well as externally, so I just don't understand why this would happen. Naturally, I am fairly new to Radioss/OpenRadioss...

I would appreciate anything anyone can tell me about this. Has anyone else seen this?
Thank you

**Comment by elequiniou** [comment]
Dear Marco,

Thanks a lot for your interest in OpenRadioss and Radioss ! 
This is a very good question I'll try to answer shortly here. But I also recommend to listen to my presentation from last OpenRadioss Users Day in Aachen: https://openradioss.org/presentations-aachen270623/
I'll also address this point during the next OpenRadioss Users Day which is approaching quickly, on September 26th in Philadelphia !

In short there is no big difference between Radioss and OpenRadioss in term of capabilities, excepted, few things like the encryption and also the license check only active in Radioss commercial.
However, there is one major difference, the versioning is only available for Radioss commercial. OpenRadioss is continuously updated, improved by new contributions. But also sometime some regressions might be introduced and corrected afterwards. This is probably what you encounter here (it could also come from a difference related to the compiler, depending on the platform).
In other words there is no exact correspondance between a Radioss commercial release and OpenRadios.
Periodically, we freeze some level of source code, make extensive tests, review performance, build documentation before releasing a Radioss commercial version at the same time as other Altair HyperWorks solvers.
So, if you will, with OpenRadioss, one has the advantage to access early features, as soon as they are available and contribute to the test and quality enhancement before a more mature commercial version is released.
Notice that to reduce the risk of introducing regressions, we have recently changed the mechanism to build OpenRadioss binaries, please check this news : https://github.com/orgs/OpenRadioss/discussions/2806

I would recommend you to test latest stable build and if the error is still there, then, please, try to share a reproducer of the failing example in order to have some developer looking at it.

Kind Regards,

Eric

**Comment by marcoserra68** [comment]
Dear Eric,

Thank you for your answer. I did go look at your presentation and I am registered to attend this year's open day remotely. I also downloaded the latest stable executable and tried running the model (explicit parts) but had no success. So, I have attached a reduced model that I have verified runs on Radioss (4 cores) but could not get to run on the latest OpenRadioss. You should know that eventually I would need to run the whole analysis as an implicit analysis because of the long simulation time. This should be ok also because the velocities involved are very low. I am trying to simulate a string of tools running in/out of an oil well. This is something I was running with LS-Dyna before I decided to switch to Radioss/OpenRadioss. The model I have attached is a much reduced version of the model I would actually be running.

The model is currently run in 3 phases:
- The first (0-GENE-TestModelShort...) generates the tool and well geometry, forming it in place by imposing displacements.
- The second (0-INIT-TestModelShort_0001...) first initializes the problem by freeing the imposed displacements of the tools while fixing the well and introducing contact and gravity. In this case it is run explicitly, but ideally this would be implict too.
- The third (0-INIT-TestModelShort_0002...) starts to extract the string from the well. This is currently running as an explicit step because I have not yet identified a set of parameters that will run it implicitly successfully.

The model is generated by a script. In Radioss, all 3 phases run out of the box. In OpenRadioss I cannot get it to run. I have tried both running on a 2024 version which I compiled to include the MUMPS solver as well as making it all explicit and running on the latest stable executable. I would be immensely grateful if someone could help me understand why this is.

Best regards
Marco

[TestModelShort.tar.zip](https://github.com/user-attachments/files/16890150/TestModelShort.tar.zip)

**Reply by elequiniou** [reply]
Thanks a lot Marco for taking the time to build a reproducer, this is really helpful. We will investigate shortly and come back to you.
And really great that you are registered for the user day!
Kind Regards,
Eric

**Reply by RevolT87** [reply]
Hi @marcoserra68,
I hope you are doing well :)
The issue seems to be due to the fact that you don't have MUMPS by default in OpenRadioss. MUMPS is mandatory to use implicit. This is something we already encountered: https://github.com/orgs/OpenRadioss/discussions/1693 
And here is a discussion that explains how to get MUMPS: https://github.com/orgs/OpenRadioss/discussions/485.

Maybe a workarround as you have access to altair radioss would be to run the implicit phase in Altair Radioss and then use OpenRadioss once the sta file is generated to run the explicit phases. I am saying that if it is painful for you to add MUMPS.

I hope it is helpful.
Regards,
Mathis

**Reply by marcoserra68** [reply]
Hi Mathis,
Welcome back.

I did compile OpenRadioss with MUMPS, and I was able to run one of the example problems successfully (the squashed rubber ring problem), so I know the compilation was successful. However, I tried running the model I have attached here, that runs just fine on Radioss, on my OpenRadioss with MUMPS as well as on the latest stable executable (only as an explicit problem), as suggested by Eric, but the model won't run at all.

I have even tried what you suggest - started the problem on Radioss and then transferred to OpenRadioss for the explicit phase, but this was also unsuccessful.

I suspect there is something in the input that causes a problem for OpeRadioss, but since on Radioss it runs just fine, I have no clue as to what it might be. I have looked through the .out files, and until the crash they are identical. No warning or error messages that indicate a problem.

Best regards
Marco

**Comment by marcoserra68** [comment, marked as answer]
Since I got a resolution to this problem from Mathis, I thought I'd record it here in case anyone else unfamiliar with the problem is looking for an answer.

I was trying to run a model consisting of beams and shells. Consequently, I requested a number of results including /H3D/ELEM/TENS/STRAIN. It turns out that while this produces no problem in Radioss, it does result in an issue in OpenRadioss. As soon as I removed this request, at Mathis's indication, the model ran also on OpenRadioss using the MUMPS solver.

Bottom line of Mathis's advice - if the run stops immediately, go look at the list of H3D or ANIM requests and identify if one of these requests is causing the problem. It's not always the issue, but it is an easy one to find if it is there.

Thank you Mathis.

**Comment by elequiniou** [comment]
Thanks Marco for the confirmation and appreciated feedback and thanks Mathis for the support! The dev team is looking at the specific issue with H3D option and will provide a fix later on.
